### ISSUE TITLE
Run `--invert_hash` from vw `--readable_model` output file

### ISSUE THREAD

**Issue by xinofekuator** [VowpalWabbit/vowpal_wabbit#3174]
## Short description

According to the documentation `--invert_hash` is computationally expensive, so using `--readable_model` seems more efficient for training when training data sizes are big.

It would be useful that we could get the human readable names from the output that we get from  `--readable_model`. Currently the only way to get it is to re-train the model with the same data and same options with `--invert_hash`.

Ideally, we could train the model efficiently and faster with `--readable_model` flag and then get the human readable values from that file for model understanding, which is usually not required during training, but nice to do at a later stage.

## **How this suggestion will help you/others**

No need to re-train a model when you have used  `--readable_model` to get the human-readable feature names.

**Comment by olgavrou** [comment]
Hi @xinofekuator thanks for you feature request

The reason `--readable_model` is faster than `--invert_hash` is that we don't need to (a) keep around and (b) store entire strings of the features, but only their hash which is part of the model anyway. So having `--readable_model` also store the full feature strings for potential later use will make it slow

would `--audit_regressor` achieve what you are looking for? (see [here](https://github.com/VowpalWabbit/vowpal_wabbit/wiki/Command-Line-Arguments#weight-options))

**Comment by xinofekuator** [comment]
Yes, that looks like it might be what I was looking for, thanks!
I will give it a try.

**Comment by JohnLangford** [comment]
Closing this issue as the question was resolved. The `--audit_regressor` option provides the functionality the user was looking for, allowing them to get human-readable feature names without retraining with `--invert_hash`. The user confirmed this solution works for their use case.
